File: tools/test_plug_provider_layout.zig
const std = @import("std");
const test_provider = @import("zig-vst3").pluginterfaces.vst.ivsttestplugprovider;

pub fn main() !void {
    const stdout = std.fs.File.stdout().deprecatedWriter();
    try printType(stdout, "ITestPlugProvider", test_provider.ITestPlugProvider);
    try printType(stdout, "ITestPlugProvider2", test_provider.ITestPlugProvider2);

    try printTuid(stdout, "ITestPlugProvider", test_provider.itest_plug_provider_iid);
    try printTuid(stdout, "ITestPlugProvider2", test_provider.itest_plug_provider2_iid);
}

fn printType(writer: anytype, comptime name: []const u8, comptime Type: type) !void {
    try writer.print("{s} size {} align {}\n", .{ name, @sizeOf(Type), @alignOf(Type) });
}

fn printTuid(writer: anytype, comptime name: []const u8, bytes: [16]u8) !void {
    try writer.print("{s} iid", .{name});
    for (bytes) |byte| {
        try writer.print(" {X:0>2}", .{byte});
    }
    try writer.writeByte('\n');
}

File: tests/abi/test_plug_provider_layout.cpp
#include "pluginterfaces/vst/ivsttestplugprovider.h"

#include <cstdio>

template <typename T>
static void print_iid (const char* name, const T& tuid)
{
	std::printf ("%s iid", name);
	for (int index = 0; index < 16; ++index)
		std::printf (" %02X", static_cast<unsigned char> (tuid[index]));
	std::printf ("\n");
}

int main ()
{
	print_iid ("ITestPlugProvider", Steinberg::Vst::ITestPlugProvider_iid);
	print_iid ("ITestPlugProvider2", Steinberg::Vst::ITestPlugProvider2_iid);
	return 0;
}
